internal/pkg/cli/storage_init.go

```diff
@@ -172,23 +308,180 @@ func (o *initStorageOpts) askStorageName() error {
 	if err != nil {
 		return fmt.Errorf("input storage name: %w", err)
 	}
-	o.StorageName = name
+	o.storageName = name
 	return nil
 }
 
 func (o *initStorageOpts) askStorageSvc() error {
-	if o.StorageSvc != "" {
+	if o.storageSvc != "" {
 		return nil
 	}
-	localSvcNames, err := o.ws.ServiceNames()
+	svc, err := o.sel.Service(storageInitSvcPrompt,
+		storageInitSvcHelp,
+		o.AppName(),
+	)
 	if err != nil {
 		return fmt.Errorf("retrieve local service names: %w", err)
 	}
-	svc, err := o.prompt.SelectOne(storageInitSvcPrompt,
-		storageInitSvcHelp,
-		localSvcNames,
+	o.storageSvc = svc
+	return nil
+}
+
+func (o *initStorageOpts) askDynamoPartitionKey() error {
+	if o.partitionKey != "" {
+		return nil
+	}
+	keyPrompt := fmt.Sprintf(fmtStorageInitDDBKeyPrompt,
+		color.HighlightUserInput("partition key"),
+		color.HighlightUserInput(dynamoDBStorageType),
+	)
+	key, err := o.prompt.Get(keyPrompt,
+		storageInitDDBPartitionKeyHelp,
+		dynamoTableNameValidation,
 	)
-	o.StorageSvc = svc
+	if err != nil {
+		return fmt.Errorf("get DDB partition key: %w", err)
+	}
+
+	keyTypePrompt := fmt.Sprintf(fmtStorageInitDDBKeyTypePrompt, ddbKeyString)
+	keyTypeHelp := fmt.Sprintf(fmtStorageInitDDBKeyTypeHelp, ddbKeyString)
+
+	keyType, err := o.prompt.SelectOne(keyTypePrompt,
+		keyTypeHelp,
+		attributeTypesLong,
+	)
```

Comment:
For the prompts, can we use the new `WithFinalMessage` option, like here:
https://github.com/aws/amazon-ecs-cli-v2/blob/b7b7f22230bca90cf8d89fd124a13772e29a5b59/internal/pkg/cli/svc_init.go#L275-L276

Usually the final message is set to the string that was emphasized in the prompt and it makes it look v. nice by having succinct prompts after the user answers questions.